app/models/rss_log.rb

```diff
@@ -444,4 +443,38 @@ def self.escape(str)
   def self.unescape(str)
     str.to_s.gsub(/%(..)/) { Regexp.last_match(1).hex.chr }
   end
+
+  #############################################################################
+
+  private
+
+  def number_of_objects_changed?(tag, time)
+    target_simply_destroyed? ||
+      target_combined?(tag) ||
+      target_recently_created?(time)
+  end
+
+  def target_simply_destroyed?
+    !target_type
+  end
+
+  def target_combined?(tag)
+    !target_id || tag.to_s.match?(/^log_#{target_type}_(merged|destroyed)/)
+  end
+
+  def target_recently_created?(time)
+    !time || time < created_at + 1.minute
+  end
+
+  def creation_detail(args)
+    if [:observation, :species_list].include?(target_type)
+      :rss_created_at.t(type: target_type) # user would be redundant
+    else
+      begin
+        "#{:rss_created_at.t(type: target_type)} by #{args[:user]}"
```

Comment:
Can we really expect args[:user] to be the object's creator?  I suppose it usually will be true.  But is this reliable?  Actually the more likely case would be args[:user] not being set at all for whatever the latest log entry is.  Maybe I'm creating bugs where bugs don't exist...

Dude!  Just got a flash of inspiration.  We are already parsing the whole log!  But we're throwing away everything but the latest entry.  Then we discover that we actually want the last entry, but instead of just keeping the entire parsed log, we choose to kludge the hell out of it.  This makes no sense.  How about this instead:

  def detail
    parse = parse_log
    latest_tag, latest_args, latest_time = parse.first
    if !target_type
      :rss_destroyed.t(type: :object)
    elsif target_combined?(latest_tag)
      :rss_destroyed.t(type: target_type)
    elsif target_created_recently?(latest_time)
      log_message(*parse.last)
    else
      log_message(*parse.first)
    end
  rescue StandardError
    ""
  end

  def log_message(tag, args, time)
    tag.t(args)
  end